### PR TITLE
added book service

### PR DESCRIPTION
**Pull request by RomanYuzviak**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false)
+    private String isbn;
```
Comment (Parashchuk) [inline src/main/java/com/example/onlinebookstore/model/Book.java:24]: it's have to be a unique value

**Review by Parashchuk — APPROVED** [review]
LGTM

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/onlinebookstore/model/Book.java:14]: Lombok is a great library but be [attentive when using it with Entity](https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/) JFYI
Also let’s  add Table annotation

File: src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java]: This annotation is redundant (more details in spring documentation)

File: src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception ex) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not create the book: " + book, ex);
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Let’s  add custom DataProcessingException

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import com.example.onlinebookstore.service.BookService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java]: the same

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few mistakes;)

File: src/main/java/com/example/onlinebookstore/model/Book.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/com/example/onlinebookstore/model/Book.java:14]: Not fixed. 
Need to add Table annotation

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! 
Please, add Table annotation!

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**